File: backoffice/tests/shared/IntValueObjectTest.py
"""
 *
 * Libraries 
 *
"""

from unittest          import TestCase
from src.shared.domain import IntValueObject

"""
 *
 * Class
 *
"""

class IntValueObjectTest( TestCase ):

    """
     *
     * Methods 
     *
    """

    def testIsNotAnEqualObject( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.equals( IntValueObject( 4 ) )
        self.assertFalse( response )
    
    def testIsAnEqualObject( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.equals( IntValueObject( 3 ) )
        self.assertTrue( response )

    def testIsAnObjectLessThan( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThan( IntValueObject( 4 ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectLessThan( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThan( IntValueObject( 3 ) )
        self.assertFalse( response )
    
    def testIsNotAnObjectLessThan2( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThan( IntValueObject( 2 ) )
        self.assertFalse( response )
    
    def testIsAnObjectLessThanOrEqual( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThanOrEqual( IntValueObject( 3 ) )
        self.assertTrue( response )

    def testIsAnObjectLessThanOrEqual2( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThanOrEqual( IntValueObject( 4 ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectLessThanOrEqual( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isLessThanOrEqual( IntValueObject( 2 ) )
        self.assertFalse( response )
    
    def testIsAnObjectGreaterThan( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 4 )
        response    = valueObject.isGreaterThan( IntValueObject( 3 ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectGreaterThan( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isGreaterThan( IntValueObject( 3 ) )
        self.assertFalse( response )
    
    def testIsNotAnObjectGreaterThan2( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 2 )
        response    = valueObject.isGreaterThan( IntValueObject( 3 ) )
        self.assertFalse( response )
    
    def testIsAnObjectGreaterThanOrEqual( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 3 )
        response    = valueObject.isGreaterThanOrEqual( IntValueObject( 3 ) )
        self.assertTrue( response )

    def testIsAnObjectGreaterThanOrEqual2( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 4 )
        response    = valueObject.isGreaterThanOrEqual( IntValueObject( 3 ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectGreaterThanOrEqual( self ) -> None:
        # Variables
        valueObject : IntValueObject
        response    : bool
        # Code
        valueObject = IntValueObject( 2 )
        response    = valueObject.isGreaterThanOrEqual( IntValueObject( 3 ) )
        self.assertFalse( response )

File: preparation/tests/shared/IntValueObjectTest.cs
/* 
 * 
 * Libraries
 *
*/

using NUnit.Framework;
using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.tests.shared {

    [TestFixture]
    public class IntValueObjectTest {

        /* 
         * 
         * Methods
         *
        */

        [Test]
        public void testIsNotAnEqualObject() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.equals( new IntValueObject( 4 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnEqualObject() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.equals( new IntValueObject( 3 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsAnObjectLessThan() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThan( new IntValueObject( 4 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectLessThan() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThan( new IntValueObject( 3 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsNotAnObjectLessThan2() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThan( new IntValueObject( 2 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnObjectLessThanOrEqual() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThanOrEqual( new IntValueObject( 3 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsAnObjectLessThanOrEqual2() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThanOrEqual( new IntValueObject( 4 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectLessThanOrEqual() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isLessThanOrEqual( new IntValueObject( 2 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnObjectGreaterThan() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 4 );
            response    = valueObject.isGreaterThan( new IntValueObject( 3 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectGreaterThan() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isGreaterThan( new IntValueObject( 3 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsNotAnObjectGreaterThan2() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 2 );
            response    = valueObject.isGreaterThan( new IntValueObject( 3 ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnObjectGreaterThanOrEqual() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 3 );
            response    = valueObject.isGreaterThanOrEqual( new IntValueObject( 3 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsAnObjectGreaterThanOrEqual2() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 4 );
            response    = valueObject.isGreaterThanOrEqual( new IntValueObject( 3 ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectGreaterThanOrEqual() {
            // Variables
            IntValueObject valueObject;
            bool           response;
            // Code
            valueObject = new IntValueObject( 2 );
            response    = valueObject.isGreaterThanOrEqual( new IntValueObject( 3 ) );
            Assert.IsFalse( response );
        }
    
    }

}